tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenAssertDifferentLsidOnLastTwoCommands.cs

```diff
@@ -0,0 +1,58 @@
+﻿/* Copyright 2019-present MongoDB Inc.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+using FluentAssertions;
+using MongoDB.Bson;
+using MongoDB.Driver.Core;
+using MongoDB.Driver.Core.Events;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading;
+
+namespace MongoDB.Driver.Tests.JsonDrivenTests
+{
+    public class JsonDrivenAssertDifferentLsidOnLastTwoCommands : JsonDrivenTestRunnerTest
+    {
+        private readonly EventCapturer _eventCapturer;
+
+        public JsonDrivenAssertDifferentLsidOnLastTwoCommands(IJsonDrivenTestRunner testRunner, EventCapturer eventCapturer, Dictionary<string, object> objectMap)
+            : base(testRunner, objectMap)
+        {
+            _eventCapturer = eventCapturer;
+        }
+
+        public override void Act(CancellationToken cancellationToken)
+        {
+            // do nothing
+        }
+
+        public override void Assert()
+        {
+            var lastTwoCommands = _eventCapturer
+                .Events
+                .Skip(_eventCapturer.Events.Count - 2)
+                .Select(commandStartedEvent => ((CommandStartedEvent)commandStartedEvent).Command);
+
+            AssertDifferentLsid(lastTwoCommands.First(), lastTwoCommands.Last());
```

Comment:
It's not too important in this context but because LINQ is lazy the way this is coded executes the full LINQ expression twice. Compare to:

    var lastTwoCommands = _eventCapturer
        .Events
        .Skip(_eventCapturer.Events.Count - 2)
        .Select(commandStartedEvent => ((CommandStartedEvent)commandStartedEvent).Command)
        .ToList();

    AssertDifferentLsid(lastTwoCommands[0], lastTwoCommands[1]);